### ISSUE TITLE
Remove deprecated Python versions from the CI script

### ISSUE THREAD

**Issue by vsnever** [cherab/core#390]
Looks like `ubuntu-latest` switched to 22.04, and Python 3.6 is not available there. This results to `Error: Version 3.6 with arch x64 not found` when running the CI tests. Therefore, the list of Python versions must be updated in the CI script. Also, I suggest testing only with the oldest and the latest supported numpy versions for each Python version:
```
matrix:
  numpy: ["oldest-supported-numpy", "numpy"]
  python-version: ["3.7", "3.8", "3.9", "3.10"]

...
run: python -m pip install cython>=0.28 ${{ matrix.numpy }} scipy matplotlib pyopencl[pocl]==2022.1
```
@jacklovell, what do you think?

**Comment by jacklovell** [comment]
That sounds like a good idea to me. We'll continue to release wheels for 3.6 I think, as they'll be built on the manylinux platform which still supports 3.6.

We could also look at testing on a manylinux image rather than Ubuntu. Pros: test on the same platform we build PyPI wheels for, and it's a platform targeted for widest Linux compatibility anyway to passing CI on manylinux *should* avoid issues on common distros. Cons: nobody actually uses manylinux on their work machines: Ubuntu (or RHEL and clones) are more common: should we test on these systems if they're closer to the environments Cherab will actually be run on?

**Comment by vsnever** [comment]
I think ideally we should be testing on the latest and oldest supported configurations to detect earlier if new versions of packages break our code, and if our code continues to work with older packages. However, manylinux is not in the list of [GitHub hosted runners](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners). 
Maybe we should test on the latest version of Ubuntu in the last three Python environments (3.9, 3.10, 3.11) and the latest version of numpy, and also on Ubuntu 18.04 in three old Python environments (3.6, 3.7, 3.8) with the oldest supported version of numpy?

**Comment by jacklovell** [comment]
Fixed by #391
